radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -31,48 +31,35 @@
  * Manages pending votes for various vertices
  */
 public final class PendingVotes {
-	private final QuorumRequirements quorumRequirements;
 	private final HashMap<Hash, ECDSASignatures> pendingVotes = new HashMap<>();
 
-	@Inject
-	public PendingVotes(QuorumRequirements quorumRequirements) {
-		this.quorumRequirements = Objects.requireNonNull(quorumRequirements);
-	}
-
 	/**
 	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.
 	 *
-	 * The vote will only be inserted if permitted by the installed {@link QuorumRequirements}.
-	 * Similarly, a QC will only be formed if permitted by the installed {@link QuorumRequirements}.
+	 * The vote will only be inserted if permitted by the {@link ValidatorSet}.
+	 * Similarly, a QC will only be formed if permitted by the {@link ValidatorSet}.
 	 * @param vote The vote to be inserted
 	 * @return The generated QC, if any
 	 */
-	public Optional<QuorumCertificate> insertVote(Vote vote) {
+	public Optional<QuorumCertificate> insertVote(Vote vote, ValidatorSet validatorSet) {
 		Objects.requireNonNull(vote, "vote");
 
 		Hash voteId = vote.getVertexMetadata().getId();
 		ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
 		ECDSASignatures signatures = pendingVotes.getOrDefault(voteId, new ECDSASignatures());

```

Comment:
Should be using DefaultSignatures.emptySignatures() instead of fresh empty every call here.